SecondClasses/quicksort/src/quicksort.erl

```diff
@@ -1,6 +1,39 @@
 -module(quicksort).
 
 -export([sort/1]).
+-export([while/1]).
+
+sort([]) -> [];
+sort(L = [H|_]) ->
+    %sort_step_by_step(L).
+    sort_with_list_comprachentions(L).
+    %sort_with_foldl(L).
+
+% 1.1 sort_step_by_step
+sort_step_by_step([H|T]) ->
+    S = lists:filter(fun(X) -> X < H end, T), 
+    B = lists:filter(fun(X) -> X >= H end, T),
+    SortedS = lists:sort(S),
```

Comment:
Well, the idea was not to use `lists:sort/1`. I was rather considering to sort all the way down yourself like:
```erlang
quicksort([]) -> [];
quicksort([H | T]) ->
    S = lists:filter(fun(X) -> X < H end, T), 
    B = lists:filter(fun(X) -> X >= H end, T),
    quicksort(S) ++ [H] ++ quicksort(B).
```